### PR TITLE
Implementation of task 8

### LINKED ISSUE #24: [Task 8] Implement API Endpoints to Accept Resume Upload (PDF) or Pasted Text with Validation
## User story
As a user, I'd like to upload a resume in PDF format or paste a job description as text to facilitate streamlined data validation and processing.

## Description
Develop API endpoints for handling resume uploads (PDFs) and text-based job descriptions. Ensure robust validation, including file type, file size, and text length, to maintain system integrity. Provide meaningful feedback on errors and successes to enhance user experience.

## Acceptance Criteria
- [ ] An API endpoint exists for uploading resumes, which:
  - [ ] Accepts only PDF or docx files.
  - [ ] Validates the uploaded file does not exceed 2MB.
  - [ ] Responds with success or failure messages, as defined.
- [ ] An API endpoint exists for job description submission, which:
  - [ ] Accepts text input up to 5,000 characters.
  - [ ] Cleans input to remove extraneous whitespace.
  - [ ] Responds with success or failure messages, as defined.
- [ ] Unit tests for the resume upload endpoint:
  - [ ] Ensure successful uploads of valid files (PDF or docx under 2MB).
  - [ ] Reject invalid file types with proper error messages.
  - [ ] Reject oversized files with proper error messages.
- [ ] Unit tests for the job description text input endpoint:
  - [ ] Ensure successful submissions of valid text (under 5,000 characters).
  - [ ] Reject text exceeding the character limit with proper error messages.
- [ ] Unit tests confirm appropriate cleaning of extraneous whitespace in text inputs.

### PR DESCRIPTION
**Pull request by RCruz02**
Broke task 8 into 2 separate pieces, jobDescription.js, and resumeUpload.js, and create tests for each.

### CODE REVIEW

File: tests/jobDescription.test.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      .post("/api/job-description")
+      .send({
+        job_description: "",
+      });
+
+    expect(response.status).toBe(400);
+    expect(response.body.error).toBe("Job description text is required.");
+  });
+
+  it("should clean job descriptions with extra whitespace", async () => {
```
Comment (truongd3) [inline tests/jobDescription.test.js:45]: Good test example!

File: backend/index.js [hunk 2]
```diff
@@ -1,5 +1,15 @@
+const express = require("express");
 const app = require('./userSignUp');
 const { extractTextFromPdf } = require("./pdfExtractor");
+const jobDescriptionRoutes = require("./jobDescription")
+const resumeUploadRoutes = require("./resumeUpload");
+
+const app = express();
```
Comment (truongd3) [inline backend/index.js:7]: Error. @danhelmer good catch!

**Review by truongd3 — CHANGES_REQUESTED** [review]

**Comment by RCruz02** [comment]
Screenshot 2024-11-25 at 1 09 03 PM

Screenshot 2024-11-25 at 1 09 18 PM

Screenshot 2024-11-25 at 1 09 30 PM

**Comment by RCruz02** [comment]
Screenshot 2024-11-25 at 1 11 39 PM

Screenshot 2024-11-25 at 1 15 22 PM

Screenshot 2024-11-25 at 1 16 38 PM

File: backend/index.js [hunk 3]
```diff
@@ -1,5 +1,15 @@
-const app = require('./userSignUp');
-const { extractTextFromPdf } = require("./pdfExtractor");
+const express = require("express");
```
Comment (truongd3) [inline backend/index.js:2]: Insert `const cors = require("cors");

**Review by fchen829 — APPROVED** [review]

**Review by truongd3 — APPROVED** [review]
lgtm now

**Pull request merged**